### DISCUSSION TITLE
Ente Auth is not correctly calculating 8-digit code

### DISCUSSION THREAD

**Discussion by ewleonardspock** [ente-io/ente#3019] [Q&A]
One of my services uses an 8-digit code (rather than the normal 6-digit codes) and the code Ente Auth is displaying is not being accepted by the service, and also doesn’t match the code shown in other authenticator apps.

**Comment by Jetspectre** [comment]
It would be nice if this gets implemented similar to Aegis's Halves Code Digit Grouping

**Comment by ewleonardspock** [comment]
This can be closed, I discovered that Google Authenticator must not have exported something correctly, because I was able to go back and recreate the code and now Ente is working as expected.

**Comment by Pharrcyde** [comment]
I believe this is still an issue because when I manually copied the secret key from a service that uses 8-digit codes, it did not save correctly and provide an 8-digit code. To verify, I took a screenshot of the QR code and sent it to another device. Scanning the QR code on that device saved it correctly and displayed 8 digits.

When scanning QR codes from a service, they contain data indicating the code length. In my case, it was:
otpauth://totp/MyService:MyUsername?secret=MySecretCode&issuer=MyIssuer&algorithm=SHA1&digits=8&period=30

However, when entering details manually with just the secret code, there is no option to change the code length to 8. For comparison, other authenticator apps like 2FAS allow changing it to 8 in the advanced settings when adding manually with a secret key.

I have searched for a similar setting in this app but couldn't find one. 
[Feature Request to add 8 digit option when adding manually](https://github.com/ente-io/ente/discussions/3174#discussion-7157928)
